typeDefs.js

```diff
@@ -381,6 +385,10 @@ const typeDefs = gql`
     addComment(productId: ID!, comment: commentInput!): CommentResult
     deleteComment(id: ID!): CommentResult
     updateComment(id: ID!, comment: commentInput!): CommentResult
+
+    "Rate Mutation"
+    addRate(product: ID!, userRate: UserRateInput!): Product
+    updateRate(product: ID!, userRate: UserRateInput!): Product
```

Comment:
It should return the `ProductResult` because it can contain an error if the product is not found

Reply:
changed